apps/frontend/core-dapp/src/stores/entities/MarketEntity.ts

```diff
@@ -286,6 +305,32 @@ export class MarketEntity
     return getContractCall(this.getCeilingValuation())?.toNumber()
   }
 
+  get finalLongPrice(): BigNumber | undefined {
+    const { maxPrice } = this
+    const finalLongPrice = getContractCall(this.getFinalLongPrice())
+
+    if (maxPrice === undefined || finalLongPrice === undefined) return undefined
+
+    const { overrideFinalLongPayout } = this.root.debugStore
+
+    if (overrideFinalLongPayout !== undefined) {
+      return maxPrice.mul(overrideFinalLongPayout).div(100)
+    }
+
+    if (
+      // The smart contract initially sets finalLongPrice to maxPrice + 1
+      // To indicate a null value (since 0 could be a valid finalLongPrice)
+      finalLongPrice.gt(maxPrice)
+    )
+      return undefined
```

Comment:
minor comment to make this shorter and more readable

```suggestion
    // The smart contract initially sets finalLongPrice to maxPrice + 1
    // To indicate a null value (since 0 could be a valid finalLongPrice)
    if (finalLongPrice.gt(maxPrice)) return undefined
```